### PR TITLE
Full Host boot disk uses PXELinux for iso template.

### PR DESCRIPTION
**Pull request by jcpunk**
The Full Host boot disk uses the PXELinux for iso the template.  If the network boot mode is Static, then the network information needs to be passed into kickstart so the interface can be brought up.

Without this patch, using the Full Host boot disk on a network without DHCP is not possible.

Basically, a rebased version of #289 

I added in the fixes for the GRUB versions as well.

### CODE REVIEW

**Comment by mmoll** [comment]
could anybody test this? @lzap? @ares? @stbenjam?

**Comment by lzap** [comment]
Hello @jcpunk and thanks for the contribution!

I am getting syntax error in safemode, can you test it please with safemode turned on?

```
safemode (1.2.4) lib/safemode.rb:61: syntax error, unexpected ',', expecting ')'
....to_jail << "ks.device=bootif", "ksdevice=bootif", "BOOTIF=#...
...                               ^
safemode (1.2.4) lib/safemode.rb:61: syntax error, unexpected ',', expecting ')'
...ice=bootif", "ksdevice=bootif", "BOOTIF=#{bootif}")
...                               ^
safemode (1.2.4) lib/safemode.rb:62: syntax error, unexpected ',', expecting ')'
(options.to_jail << "inst.ks.sendmac", "ks.sendmac", "kssendmac")
                                      ^
safemode (1.2.4) lib/safemode.rb:62: syntax error, unexpected ',', expecting ')'
(options.to_jail << "inst.ks.sendmac", "ks.sendmac", "kssendmac")
                                                    ^
safemode (1.2.4) lib/safemode.rb:70: syntax error, unexpected ',', expecting ')'
...(options.to_jail << "ip=#{ip}", "netmask=#{mask}", "gateway=...
...                               ^
safemode (1.2.4) lib/safemode.rb:70: syntax error, unexpected ',', expecting ')'
... "ip=#{ip}", "netmask=#{mask}", "gateway=#{gw}", "dns=#{dns}...
...                               ^
safemode (1.2.4) lib/safemode.rb:70: syntax error, unexpected ',', expecting ')'
...mask=#{mask}", "gateway=#{gw}", "dns=#{dns}")
...                               ^
safemode (1.2.4) lib/safemode.rb:72: syntax error, unexpected ',', expecting ')'
...p=#{ip}::#{gw}:#{mask}:::none", "nameserver=#{dns}")
...                               ^
safemode (1.2.4) lib/safemode.rb:51:in `eval'
```

It's the PXELinux one I started testing with.

**Comment by jcpunk** [comment]
I'll see what I can do.  I've got safe mode `off` for some other, more dangerous things, I've got going on that are (sadly) needed in production.  I may have to spin up a test instance of foreman....

I don't suppose there is a way I can enable safe mode for a single template is there?  Or a command line tool I could run against the erb?

**Comment by lzap** [comment]
Yeah bummer, the big plan is to add tests in this repo, but we only have few which are not related. If you'd write simple tests for these via Rails ERB (do not use Ruby ERB that's a different beast) and SafeMode that would be fantastic.

**Comment by jcpunk** [comment]
Alas, I haven't a clue how to write those sorts of tests...  My ERB experience is mostly confined to googling syntax for various puppet things.

**Comment by jcpunk** [comment]
I spun up a quick forman image and was able to get the template to render in safemode with a few changes.

**Comment by lzap** [comment]
Testing so far with CentOS 7.1:

```
# This file was deployed via 'Kickstart default PXELinux' template


TIMEOUT 10
DEFAULT Kickstart default PXELinux

LABEL Kickstart default PXELinux
  KERNEL boot/RedHat-7.1-x86_64-vmlinuz
  APPEND initrd=boot/RedHat-7.1-x86_64-initrd.img ks=http://192.168.122.1:8448/unattended/provision?token=81abef15-97e7-45e9-afb9-e160c1b0b455  network BOOTIF=00-52-54-00-fc-3d-4d ks.device=bootif ks.sendmac inst.ks.sendmac
  IPAPPEND 2
```

```
# This file was deployed via 'Kickstart default PXEGrub' template


default=0
timeout=10

title Kickstart default PXEGrub
  root (nd)
  kernel (nd)/../boot/RedHat-7.1-x86_64-vmlinuz ks=http://192.168.122.1:8448/unattended/provision?token=52e002d5-2dca-4415-84e7-99bf7497d275  network BOOTIF=00-52-54-00-47-5b-53 ks.device=bootif ks.sendmac inst.ks.sendmac
  initrd (nd)/../boot/RedHat-7.1-x86_64-initrd.img
```

```
# This file was deployed via 'Kickstart default PXEGrub2' template


set default=0
set timeout=10

menuentry 'Kickstart default PXEGrub2' {
  linuxefi boot/RedHat-7.1-x86_64-vmlinuz ks=http://192.168.122.1:8448/unattended/provision?token=52e002d5-2dca-4415-84e7-99bf7497d275  network BOOTIF=00-52-54-00-47-5b-53 ks.device=bootif ks.sendmac inst.ks.sendmac
  initrdefi boot/RedHat-7.1-x86_64-initrd.img
}
```

It renders fine, I will do some provisioning tomorrow.

File: kickstart/PXEGrub.erb [hunk 1]
```diff
@@ -8,18 +8,63 @@ oses:
...
+    bootif = '00-' + mac.gsub(':', '-')
+    options.push("BOOTIF=#{bootif}")
+  end
+
+  # Tell Anaconda what to pass off to kickstart server
+  if (@host.operatingsystem.name.match(/Fedora/i) && major < 17) || major < 7
+    options.push("kssendmac", "ksdevice=bootif")
+  else
+    options.push("ks.device=bootif", "ks.sendmac", "inst.ks.sendmac")
+  end
```
Comment (lzap) [inline kickstart/PXEGrub.erb]: Can you shorten this to:

    options.push("ks.device=bootif", "kssendmac", "ks.sendmac", "inst.ks.sendmac")

The if statement is no longer necessary, it's better to provide all three possible variations of this Anaconda flag. Do this in all templates please.

File: kickstart/PXEGrub.erb [hunk 2]
```diff
@@ -8,18 +8,63 @@ oses:
 -%>
 # This file was deployed via '<%= template_name %>' template
 
+<%
+  major = @host.operatingsystem.major.to_i
+  mac = @host.provision_interface.mac
+
+  # Tell Anaconda to perform network functions with boot interface
+  options = ["network",]
```
Comment (lzap) [inline kickstart/PXEGrub.erb]: Why the comma?

Comment (jcpunk) [reply kickstart/PXEGrub.erb]: Found that in a Best Practices ERB guide.  Guess it isn't actually Best Practices....

File: kickstart/PXEGrub.erb (hunk 1 above)
Comment (jcpunk) [reply kickstart/PXEGrub.erb]: Done, I'll squash and repush

**Comment by jcpunk** [comment]
Requested fixes in place

**Comment by jcpunk** [comment]
Just checking back in on this

**Comment by mmoll** [comment]
@lzap did you test provisioing? ☝️

**Comment by lzap** [comment]
Sorry for the delay, testing.

**Comment by lzap** [comment]
Thanks works fine.
